advanced/src/app/core/authentication/authentication.service.ts

```diff
@@ -10,31 +11,36 @@ import { Router } from '@angular/router';
 @Injectable()
 export class AuthenticationService {
   auth0 = new auth0.WebAuth({
-    clientID: '<client-id>',
-    domain: '<domain>',
+    clientID: environment.auth0ClientId, // '<client-id>',
+    domain: environment.auth0Domain, // '<domain>',
     responseType: 'token id_token',
-    redirectUri: '<redirect-uri>',
+    redirectUri: environment.auth0RedirectUri, // '<redirect-uri>',
     scope: 'openid'
   });
 
   constructor(private router: Router) {}
 
+  /** Authorizes with Auth0 */
+  public login(): void {
+    this.auth0.authorize();
+  }
+
+  /** Logs out by clearing the access and ID tokens */
   public logout(): void {
```

Comment:
How about changing this to the logout method provided by Auth0. That means adding the method. 
```
this.auth0.logout({
      returnTo: window.location.origin
    });
```